pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/performance/UseStringBufferForStringAppendsRule.java

```diff
@@ -66,13 +77,29 @@ public Object visit(ASTVariableDeclaratorId node, Object data) {
                         ASTAssignmentOperator assignmentOperator = statement
                                 .getFirstDescendantOfType(ASTAssignmentOperator.class);
                         if (assignmentOperator != null && assignmentOperator.isCompound()) {
-                            addViolation(data, assignmentOperator);
+                            //CS304 Issue link: https://github.com/pmd/pmd/issues/1736
+                            // check whether is first time to break the rule
+                            if (!map.containsKey(astName.getNameDeclaration().getName())) {
```

Comment:
Please extract `astName.getNameDeclaration().getName()` into a local variable. You use it 8 times. If I read the code correctly, that's actually the variable's name.